### PR TITLE
Added Stacking to auto-sklearn

### PR DESCRIPTION
**Pull request by zemoel**

### CODE REVIEW

File: autosklearn/ensemble_selection_script.py [hunk 1]
```diff
@@ -266,9 +268,16 @@ def main(autosklearn_tmp_dir,
             continue
 
         else:
-            ensemble = EnsembleSelection(ensemble_size=ensemble_size,
+            if ensemble_method == "EnsembleSelection":
+                ensemble = EnsembleSelection(ensemble_size=ensemble_size,
```
Comment (Borda) [inline autosklearn/ensemble_selection_script.py]: line alignment, PEP8

File: autosklearn/ensembles/abstract_ensemble.py [hunk 2]
```diff
@@ -26,6 +26,18 @@ def fit(self, base_models_predictions, true_targets, model_identifiers):
         """
         pass
 
+    @abstractmethod
+    def get_train_score(self):
+        """ Reports the methods score during training.
+
+        Parameters
```
Comment (Borda) [inline autosklearn/ensembles/abstract_ensemble.py]: if there are no parameters nor return values, the headlines are not needed, right?

File: autosklearn/ensembles/stacking.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+    Class for stacked classification. Implements ideas from: "Issues in Stacked Generalization" by Kai Ming Ting, and Ian H. Witten in JAIR 99
+    """
+
+    def __init__(self, ensemble_size, task_type, metric,
+                 sorted_initialization=False, bagging=False, mode='fast'):
+        
+
+    @staticmethod
+    def compute_y(l, N, true_labels):
+
```
Comment (Borda) [inline autosklearn/ensembles/stacking.py]: remove new line

File: autosklearn/ensembles/stacking.py [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+        N: array of shape = [n_data_points]
+        true_targets : array of shape [n_data_points]
+
+
+        Returns
+        -------
+        y: array of shape = [n_base_models]
+
+        """
+        y = true_labels
```
Comment (Borda) [inline autosklearn/ensembles/stacking.py]: probably use a copy here `y = true_labels.copy()`

File: autosklearn/ensembles/stacking.py [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+        y[y != l] = 0
+        y[y == l] = 1
+
+        return y
```
Comment (Borda) [inline autosklearn/ensembles/stacking.py]: rather `return np.array(true_labels == l, dtype=bool)`

File: autosklearn/ensembles/stacking.py [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+    """
+    Class for stacked classification. Implements ideas from: "Issues in Stacked Generalization" by Kai Ming Ting, and Ian H. Witten in JAIR 99
+    """
+
+    def __init__(self, ensemble_size, task_type, metric,
+                 sorted_initialization=False, bagging=False, mode='fast'):
+        
+
+    @staticmethod
+    def compute_y(l, N, true_labels):
```
Comment (Borda) [inline autosklearn/ensembles/stacking.py]: N is not used

File: autosklearn/ensembles/stacking.py [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+            n_targets is 0 or 1 in case of regression
+
+        true_targets : array of shape [n_targets]
+
+        model_identifiers : identifier for each base model.
+            Can be used for practical text output of the ensemble.
+
+        Returns
+        -------
+        self
```
Comment (Borda) [inline autosklearn/ensembles/stacking.py]: say what is the `self` is

File: autosklearn/ensembles/stacking.py [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+        -------
+        array : [n_data_points, n_targets]
+        """
+        Z_test = np.array(base_models_predictions) 
+        L = self.L
+        K = self.K
+        N_test = Z_test.shape[1]
+        LR = np.zeros([N_test, L])        
+
+        if not((Z_test.shape[0] == K) and (Z_test.shape[2] == L)):
```
Comment (Borda) [inline autosklearn/ensembles/stacking.py]: space after not and remove redundant brackets

File: autosklearn/ensembles/stacking.py [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+            logger.error("check if predictions_test is of shape K x N_test x L")
+
+
+        for l in range(0, L):
+            LR[:, l] = np.dot(self.Alpha[:, l],Z_test[:, :, l])
```
Comment (Borda) [inline autosklearn/ensembles/stacking.py]: space after coma, PEP8

**Comment by mfeurer** [comment]
Closing this as a) the code is outdated and b) we have no reason to believe that stacking outperforms ensemble selection as implemented in Auto-sklearn.

**Pull request closed without merging**